seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/SeleniumRun.scala

```diff
@@ -129,20 +131,10 @@ private[selenium] object SeleniumRun {
       newRun: Ctor[T], failed: Throwable => T): T = {
     validator.validate(runConfig)
 
-    val scripts = input.map {
-      case Input.Script(s) => s
-      case _               => throw new UnsupportedInputException(input)
-    }
-
     try {
       withCleanup(FileMaterializer(config.materialization))(_.close()) { m =>
-        val allScriptURLs = (
-            m.materialize("setup.js", JSSetup.setupCode(enableCom)) +:
-            scripts.map(m.materialize)
-        )
-
-        val page = m.materialize("scalajsRun.html", htmlPage(allScriptURLs))
-
+        val setupJsPath = JSSetup.setupFile(enableCom)
+        val page = m.materialize("scalajsRun.html", htmlPage(setupJsPath, input, m))
```

Comment:
At this point, you can just pass this as additional input:

```scala
val fullInput = Input.Script(JSSetup.setupFile(enableCom)) :: input
/* snip */ htmlPage(fullInput, m) /* snip */
```

That'll also simplify `htmlPage`.